From bc31f7b1427d0eaa90b661ea09da6e63ec2edb7f Mon Sep 17 00:00:00 2001
From: Victor Fuentes <victor@vlinkz.dev>
Date: Sat, 26 Sep 2026 23:17:02 -0700
Subject: [PATCH 02/14] cmd: blkmap: Parse the linear target block number as
 hex

The other block numbers are hex, as is what setexpr and part start
produce, including in the documented example.

Signed-off-by: Victor Fuentes <victor@vlinkz.dev>
---
 cmd/blkmap.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmd/blkmap.c b/cmd/blkmap.c
index 65edec899..d5e5002d2 100644
--- a/cmd/blkmap.c
+++ b/cmd/blkmap.c
@@ -36,7 +36,7 @@ static int do_blkmap_map_linear(struct map_ctx *ctx, int argc,
 		return CMD_RET_USAGE;
 
 	ldevnum = dectoul(argv[2], NULL);
-	lblknr = dectoul(argv[3], NULL);
+	lblknr = hextoul(argv[3], NULL);
 
 	lbd = blk_get_devnum_by_uclass_idname(argv[1], ldevnum);
 	if (!lbd) {
